Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -0,0 +1,168 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+import Foundation
+
+extension OfflineMapAreasView {
+    /// The model class for the offline map areas view.
+    @MainActor
+    class MapViewModel: ObservableObject {
+        /// The portal item ID of the web map.
+        private var portalItemID: String?
+        
+        /// The offline map of the downloaded preplanned map area.
+        private var offlineMap: Map?
+        
+        /// The offline map task.
+        let offlineMapTask: OfflineMapTask
+        
+        /// The url for the preplanned map areas directory.
+        let preplannedDirectory: URL
+        
+        /// The mobile map packages created from mmpk files in the documents directory.
+        @Published private(set) var mobileMapPackages: [MobileMapPackage] = []
+        
+        /// The preplanned offline map information.
+        @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
+        
+        /// A Boolean value indicating whether the map has preplanned map areas.
+        @Published private(set) var hasPreplannedMapAreas = false
+        
+        /// A Boolean value indicating whether the user has authorized notifications to be shown.
+        var canShowNotifications = false
+        
+        /// The job manager.
+        let jobManager = JobManager.shared
+        
+        init(map: Map) {
+            offlineMapTask = OfflineMapTask(onlineMap: map)
+            
+            if let portalItemID = map.item?.id?.rawValue {
+                FileManager.default.createDirectories(for: portalItemID)
+                
+                preplannedDirectory = FileManager.default.preplannedDirectory(portalItemID: portalItemID)
+            } else {
+                preplannedDirectory = FileManager.default.documentsDirectory
+            }
+        }
+        
+        /// Gets the preplanned map areas from the offline map task and creates the
+        /// offline map models.
+        func makePreplannedOfflineMapModels() async {
+            preplannedMapModels = await Result {
+                try await offlineMapTask.preplannedMapAreas
+                    .sorted(using: KeyPathComparator(\.portalItem.title))
+                    .compactMap {
+                        PreplannedMapModel(
+                            preplannedMapArea: $0,
+                            offlineMapTask: self.offlineMapTask,
+                            preplannedDirectory: self.preplannedDirectory
+                        )
+                    }
+            }
+            if let models = try? preplannedMapModels?.get() {
+                hasPreplannedMapAreas = !models.isEmpty
+            }
+        }
+        
+        /// Loads locally stored mobile map packages for the map's preplanned map areas.
+        func loadPreplannedMobileMapPackages() {
```

Comment:
The term "Load" here is overloaded and makes me think of the "Loadable" pattern in the SDK. I would recommend changing this terminology.